python_modules/libraries/dagster-components/dagster_components/core/component.py

```diff
@@ -49,15 +58,27 @@ class Component(ABC):
     generate_params_schema: ClassVar = None
 
     @classmethod
-    def get_rendering_scope(cls) -> Mapping[str, Any]:
-        return {}
+    def get_generator(cls) -> Union[ComponentGenerator, ComponentGeneratorUnavailableReason]:
```

Comment:
Theoretically it might be a bit more pythonic to just raise a custom error type in the case that the generator isn't available right?

Not strongly held, I do see the benefit of forcing the system to explicitly return something of a specific type in the case that the component isn't availabl